src/test/java/com/sap/sgs/phosphor/fosstars/data/github/TestGitHubDataFetcherHolder.java

```diff
@@ -53,42 +48,89 @@ public void init() throws IOException {
   @After
   public void cleanup() throws IOException {
     List<Path> deletedPaths = new ArrayList<>();
-    fetcher.cleanup(((url, repository, total) -> {
+    fetcher.cleanup((url, repository, total) -> {
       deletedPaths.add(repository.path());
       return true;
-    }));
+    });
 
     for (Path deletedPath : deletedPaths) {
       assertFalse(Files.exists(deletedPath));
     }
 
-    FileUtils.forceDeleteOnExit(path.toFile());
+    FileUtils.forceDeleteOnExit(fetcher.baseDir().toFile());
   }
 
-  public static class TestGitHubDateFetcher extends GitHubDataFetcher {
+  public static class TestGitHubDataFetcher extends GitHubDataFetcher {
 
-    public TestGitHubDateFetcher(GitHub github, Path base) throws IOException {
-      super(github, base);
+    /**
+     * Test class constructor.
+     */
+    public TestGitHubDataFetcher(GitHub github) throws IOException {
+      super(github);
     }
 
+    /**
+     * Adds {@link GitHubProject} and its associated {@link LocalRepository} details to cache.
+     * @param project The {@link GitHubProject}.
+     * @param repository The {@link LocalRepository}.
+     */
     public void addForTesting(GitHubProject project, LocalRepository repository) {
-      localRepositories.put(project, repository);
+      LOCAL_REPOSITORIES.put(project, repository);
+    }
+
+    /**
+     * Returns the current list of local repositories stored in the cache.
+     *
+     * @return Map of local repositories in cache.
+     */
+    public Map<GitHubProject, LocalRepository> getLocalRespositoriesForTesting() {
+      return LOCAL_REPOSITORIES;
+    }
+
+    /**
+     * Returns the current list of local repositories stored in the base folder.
+     *
+     * @return Map of local repositories available.
+     * @throws IOException if something goes wrong.
+     */
+    public Map<URL, LocalRepositoryInfo> getLocalRespositoriesInfoForTesting() throws IOException {
+      return LOCAL_REPOSITORIES_INFO;
```

Comment:
`LOCAL_REPOSITORIES_INFO` has package-private access. Is this method really necessary? I see that only tests from the same package needs this field.

Reply:
I have updated this method to 
```
loadLocalRepositoriesInfo();
return LOCAL_REPOSITORIES_INFO;
```

Reply:
It's not about renaming. I mean that the method looks redundant. I think you can just remove it and use `LOCAL_REPOSITORIES_INFO` in the tests from the same package.

Reply:
I have removed this redundant method definition